Mathlib/Analysis/Normed/Unbundled/SpectralNorm.lean

```diff
@@ -644,4 +664,218 @@ theorem spectralAlgNorm_isPowMul : IsPowMul (spectralAlgNorm K L) := isPowMul_sp
 
 end IsAlgebraic
 
+end NormedField
+section NontriviallyNormedField
+
+open IntermediateField
+
+universe u v
+
+variable {K : Type u} [NontriviallyNormedField K] {L : Type v} [Field L] [Algebra K L]
+  [Algebra.IsAlgebraic K L] [hu : IsUltrametricDist K]
+
+/-- If `K` is a field complete with respect to a nontrivial nonarchimedean multiplicative norm and
+  `L/K` is an algebraic extension, then any power-multiplicative `K`-algebra norm on `L` coincides
+  with the spectral norm. -/
+theorem spectralNorm_unique [CompleteSpace K] {f : AlgebraNorm K L} (hf_pm : IsPowMul f) :
+    f = spectralAlgNorm K L := by
+  apply eq_of_powMul_faithful f hf_pm _ spectralAlgNorm_isPowMul
+  intro x
+  set E : Type v := id K⟮x⟯ with hEdef
+  letI hE : Field E := by rw [hEdef, id_eq]; infer_instance
+  letI : Algebra K E := K⟮x⟯.algebra
+  letI hidE : ZeroHomClass (id ↥K⟮x⟯ →ₗ[K] ↥K⟮x⟯) _ _ := AddMonoidHomClass.toZeroHomClass
+  set id1 : K⟮x⟯ →ₗ[K] E :=
+    { toFun := id
+      map_add' := fun x y => rfl
+      map_smul' := fun r x => rfl }
+  set id2 : E →ₗ[K] K⟮x⟯ :=
+    { toFun := id
+      map_add' := fun x y => rfl
+      map_smul' := fun r x => rfl }
+  set hs_norm : RingNorm E :=
+    { toFun := fun y : E => spectralNorm K L (id2 y : L)
+      map_zero' := by simp [map_zero, spectralNorm_zero, ZeroMemClass.coe_zero]
+      add_le' := fun a b => by
```

Comment:
```suggestion
      add_le' a b := by
```
I think mathlib prefers the style I suggest, and there are many many occurrences of this in the PR.